pkg/kubelet/server/remotecommand/exec.go

```diff
@@ -51,8 +85,19 @@ func ServeExec(w http.ResponseWriter, req *http.Request, executor Executor, podN
 
 	err := executor.ExecInContainer(podName, uid, container, cmd, ctx.stdinStream, ctx.stdoutStream, ctx.stderrStream, ctx.tty, ctx.resizeChan)
 	if err != nil {
-		msg := fmt.Sprintf("error executing command in container: %v", err)
-		runtime.HandleError(errors.New(msg))
-		fmt.Fprint(ctx.errorStream, msg)
+		m := ExitError{}
```

Comment:
That was my initial rational as well. But adding v4 is cleaner and not much more work. We have to find a way to send different error messages anyway, depending on the version.